push_docker.sh

```diff
@@ -1,42 +1,58 @@
+#! /bin/bash 
+
 DOCKER_IMAGE_FILE=$1
 REGISTRY_SERVER=$2
-REGISTRY_PORT=$3
-REGISTRY_USERNAME=$4
-REGISTRY_PASSWD=$5
-DOCKER_IMAGE_TAG=$6
+REGISTRY_USERNAME=$3
+REGISTRY_PASSWD=$4
+PLATFORM=$5
+SONIC_VERSION=""
+DOCKER_IMAGE_TAG=""
+
+shift 5
+
+if [ "$#" != "0" ]
+then
+    SONIC_VERSION=$1
+    shift
+fi
+
+if [ "$#" != "0" ]
+    DOCKER_IMAGE_TAG=$1
```

Comment:
>DOCKER_IMAGE_TAG [](start = 4, length = 16)

Originally DOCKER_IMAGE_TAG is required, why change it to optional?

Reply:
This is a repository of all dockers across all versions. All dockers are tagged with their versions. So any duplicate with additional tag can clash. For example, if you use a tag, say "latest", there can be only one, and when you upload the next version, tagging to 'latest' would fail as there exists one already.

Reply:
You did not answer my question.

---
In reply to: [402699515](https://github.com/Azure/sonic-buildimage/pull/4337#discussion_r402699515) [](ancestors = 402699515)

Reply:
Original behavior retained in the latest revision.